Average Filter

ricdezen opened the pull request:
## Premise
This pr contains a class to compute the sum and average of a set of keys in a Stream.

## Features
**feat(average_filter.py)**:
**+** Added AverageFilter class: single input and output Filter class that computes the sum and average for each atom that passes through.
**test(average_filter_test.py)** Basic tests for AverageFilter

## Picture of a cute animal
So basically I found out that [Sables](https://it.wikipedia.org/wiki/Martes_zibellina) are the cutest things ever.
![zibellino1](https://user-images.githubusercontent.com/56487105/81512829-e4346980-9323-11ea-8b79-53602c345d8f.jpg)

Commit:
feat (average_filter_test.py):
Added tests for avg/sum calculating filter

Commit:
feat (average_filter.py):
+ AverageFilter class for calculating the average and sum of a set of keys

CremaLuca commented on otri/filtering/filters/average_filter.py:
```diff
@@ -0,0 +1,69 @@
+from numbers import Number
+
+
+class AverageFilter(Filter):
```
Define inputs and outputs of filter in class specification, so it's easier and faster to link them together.

CremaLuca commented on otri/filtering/filters/average_filter.py:
```diff
@@ -0,0 +1,69 @@
+        '''
+        super().__init__([input_stream],
+                         input_streams_count=1, output_streams_count=1)
+        self.__cnt = {k: 0 for k in keys}
```
Woah didn't know you could initialise a dict like that

ricdezen commented:
Note that we might want to split this filter into two parts. A base one that only sums (which will become the sum filter) ad a subclass that also computes the average.

CremaLuca reviewed (approved):
Clean. 8/8

Commit:
Merge branch 'feat/filtering' of https://github.com/OTRI-Unipd/OTRI into feat/avg-filter

Commit:
fix (average_filter.py):
+ added check to avoid closing the output stream twice
+ Added class specs

fix (nuplicator_filter.py):
+ added check to avoid closing the output stream twice
